Projects/App/Sources/Result/ResultViewController.swift

```diff
@@ -71,21 +69,39 @@ final class ResultViewController: BaseViewController<ResultViewModelProtocol> {
 }
 
 extension ResultViewController: ResultViewButtonDelegate {
+    
     func homeButtonTapped() {
-        dismiss(animated: true)
+        navigationController?.popToRootViewController(animated: true)
     }
     
-    func saveButtonTapped() {
+    func saveButtonTapped(completion: @escaping (Bool) -> ()) {
         // TODO: 서버에서 받아온 이미지로 저장해줘야함
-        guard let image = UIImage(systemName: "pencil") else { return }
+        guard let image = UIImage(systemName: "pencil") else {
+            completion(false)
+            return
+        }
         UIImageWriteToSavedPhotosAlbum(image, self, #selector(gifticonImageSaved), nil)
+        completion(true)
     }
     
-    @objc private func gifticonImageSaved(image: UIImage, error: Error?, context: UnsafeRawPointer) {
+    func saveFailed() {
         let toastView = ToastView()
-        toastView.configureToastView(with: self.view, style: .save, image: .iconLogoCharacter)
+        toastView.configureToastView(with: self.view, style: .saveFail, image: .iconRotateLogoCharacterEmpty)
         toastView.showToastView(with: self.view)
     }
+    
+    @objc private func gifticonImageSaved(image: UIImage, error: Error?, context: UnsafeRawPointer) {
+        
+        let toastView = ToastView()
+        
+        if error == nil {
+            toastView.configureToastView(with: self.view, style: .save, image: .iconLogoCharacter)
+            toastView.showToastView(with: self.view)
+        } else {
+            toastView.configureToastView(with: self.view, style: .saveFail, image: .iconRotateLogoCharacterEmpty)
+            toastView.showToastView(with: self.view)
+        }
```

Comment:
```suggestion
        if error == nil {
            toastView.configureToastView(with: self.view, style: .save, image: .iconLogoCharacter)
        } else {
            toastView.configureToastView(with: self.view, style: .saveFail, image: .iconRotateLogoCharacterEmpty)
        }
        toastView.showToastView(with: self.view)
```

위와같이 중복코드를 제거 할 수 있을 것 같습니다! 😃

Reply:
제안해주신대로 반영하였습니다!